### PR TITLE
ControllerEmu: Allow changing the target gate radius of octagon sticks.

### PR DESCRIPTION
**Pull request by jordan-woyak**
Reimplemented feature from  #10665.

I also moved "Virtual Notches" to the new dialog.
Do we like that?

image

What should the setting be called? "Gate Size", "Gate Radius"?

### CODE REVIEW

**Comment by JMC47** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/InputCommon/ControllerEmu/Setting/NumericSetting.h [hunk 1]
```diff
@@ -20,13 +20,20 @@ enum class SettingType
   Bool,
 };
 
+enum class SettingDifficulty
```
Comment (AdmiralCurtiss) [inline Source/Core/InputCommon/ControllerEmu/Setting/NumericSetting.h:23]: 'Difficulty' seems like a strange way to phrase this, though I'm not quite sure of a better name for this either...

**Comment by AdmiralCurtiss** [comment]
Works as I expected. 'Gate Size' seems fine as a name, too. The default values are bizarre but they are what they are I suppose.

File: Source/Core/InputCommon/ControllerEmu/Setting/NumericSetting.h (hunk 1 above)
Comment (jordan-woyak) [reply Source/Core/InputCommon/ControllerEmu/Setting/NumericSetting.h:23]: How about `SettingVisibility`?

Comment (AdmiralCurtiss) [reply Source/Core/InputCommon/ControllerEmu/Setting/NumericSetting.h:23]: That works, yeah.

**Comment by Rumi-Larry** [comment]
Wouldn't it make sense to also control the thickness of the notches as well as the angle?

**Comment by jordan-woyak** [comment]
> Wouldn't it make sense to also control the thickness of the notches as well as the angle?

Maybe, but that's outside of the scope of this PR. I only moved the notches setting in the UI.

File: Source/Core/InputCommon/ControllerEmu/Setting/NumericSetting.h (hunk 1 above)
Comment (AdmiralCurtiss) [reply Source/Core/InputCommon/ControllerEmu/Setting/NumericSetting.h:23]: Are you going to make this change? I would just merge this other than this little thing.

Comment (jordan-woyak) [reply Source/Core/InputCommon/ControllerEmu/Setting/NumericSetting.h:23]: Yes. In a few hours I should get to it.

**Comment by Sage-King** [comment]
Works like I would expect it to and your code is very nice to read :)

**Review by AdmiralCurtiss — APPROVED** [review]
Seems good.

**Pull request merged**